spring-ai-alibaba-core/src/main/java/com/alibaba/cloud/ai/dashscope/rag/AnalyticdbVector.java

```diff
@@ -0,0 +1,411 @@
+               if ("1".equals(redisClientService.getValueByKey(cacheKey))) {
+                   redisClientService.deleteLock(lockName);
+                   return;
+               }
+               // Describe the collection to check if it exists
+               DescribeCollectionRequest describeRequest = new DescribeCollectionRequest()
+                       .setDBInstanceId(this.config.getDBInstanceId())
+                       .setRegionId(this.config.getRegionId())
+                       .setNamespace(this.config.getNamespace())
+                       .setNamespacePassword(this.config.getNamespacePassword())
+                       .setCollection(this.collectionName);
+               try {
+                   this.client.describeCollection(describeRequest);
+               } catch (TeaException e) {
+                   if (e.getStatusCode() == 404) {
+                       // Collection does not exist, create it
+                       String metadata = JSON.toJSONString(new JSONObject()
+                               .fluentPut("refDocId", "text")
+                               .fluentPut("content", "text")
+                               .fluentPut("metadata", "jsonb"));
+
+                       String fullTextRetrievalFields = "content";
+                       CreateCollectionRequest createRequest = new CreateCollectionRequest()
+                               .setDBInstanceId(this.config.getDBInstanceId())
+                               .setRegionId(this.config.getRegionId())
+                               .setManagerAccount(this.config.getManagerAccount())
+                               .setManagerAccountPassword(this.config.getManagerAccountPassword())
+                               .setNamespace(this.config.getNamespace())
+                               .setCollection(this.collectionName)
+                               .setDimension(embeddingDimension)
+                               .setMetrics(this.config.getMetrics())
+                               .setMetadata(metadata)
+                               .setFullTextRetrievalFields(fullTextRetrievalFields);
+                       this.client.createCollection(createRequest);
+                   } else {
+                       throw new RuntimeException("Failed to create collection " + this.collectionName + ": " + e.getMessage());
+                   }
+                  // Set the cache key to indicate the collection exists
+                  redisClientService.setKeyValue(cacheKey, "1", 3600);
+               }
+           }
+        }catch(Exception e){
+            redisClientService.deleteLock(lockName);
+            throw new RuntimeException("Failed to create collection " + this.collectionName + ": " + e.getMessage());
+         }finally {
+            redisClientService.deleteLock(lockName);
+        }
+    }
+
+    public void create(List<Document> texts, List<List<Double>> embeddings) throws Exception {
```

Comment:
This method is unnecessary, we should follow the interface pattern defined in VectoreStore by spring-ai.